samples/apps/copilot-chat-app/webapp/src/redux/features/conversations/conversationsSlice.ts

```diff
@@ -32,16 +35,34 @@ export const conversationsSlice: Slice<ConversationsState> = createSlice({
             state.conversations = { [newId]: action.payload, ...state.conversations };
             state.selectedId = newId;
         },
-        updateConversation: (
+        addUserToConversation: (state: ConversationsState, action: PayloadAction<{ user: IChatUser, chatId: string }>) => {
+            const { user, chatId } = action.payload;
+            state.conversations[chatId].users.push(user);
+        },
+        /*
+        * updateConversationFromUser() and updateConversationFromServer() both update the conversations state.
+        * However they are for different purposes. The former action is for updating the conversation from the
+        * user and will be captured by the SignalR middleware and the payload will be broadcasted to all clients
+        * in the same group.
+        * The updateConversationFromUser() action is triggered by the SignalR middleware when a response is received.
```

Comment:
Still unclear why both actions are required, the middleware should be able to dispatch the same action

Reply:
I will do my best to illustrate with the sequences below:
Scenario 1: user sends a message
middleware captures the message -> update the conversation state -> broadcast the message to other users

Scenario 2: bot responds to a message
signalR receieves the event -> dispatches the response to update the conversation state

We have to create two actions for these two scenarios otherwise each bot response will create two messages in the conversation state.

Reply:
I see, if all messages come back anyway (even if you're the user that sent them), can't you skip updating the conversation on sendMessage altogether and rely on the conversation state to be updated once the message has been broadcasted to all chat participants?

Reply:
Yeah, so the getResponse function will not return a message rn. The chat will get populated when the server broadcasts the response back, including the client who sent the message.